adapter/internal/oasparser/envoyconf/routes_with_clusters.go

```diff
@@ -182,6 +182,19 @@ func CreateRoutesWithClusters(mgwSwagger model.MgwSwagger, upstreamCerts map[str
 		}
 	}
 
+	// Websocket APIs are processed in a different manner compared to REST APIs.
+	// No interceptors engaged.
+	// There is a single method, which is a GET.
+	// No topic level endpoints.
+	if mgwSwagger.GetAPIType() == constants.WS {
+		for _, resource := range mgwSwagger.GetResources() {
+			routesP := createRoute(genRouteCreateParams(&mgwSwagger, resource, vHost, apiLevelbasePath, apiLevelClusterNameProd,
+				apiLevelClusterNameSand, nil, nil, organizationID))
+			routes = append(routes, routesP)
+		}
+		return routes, clusters, endpoints
+	}
```

Comment:
Should we include the other async API types too? https://github.com/wso2/product-microgateway/blob/151b1d0ed5b8a1bb814b9ac3ef12a6684c507b59/adapter/internal/oasparser/model/api_yaml.go#L243

Reply:
Are we sure about not having topic level endpoints?

Reply:
https://github.com/wso2/product-microgateway/issues/2677
I created a separate issue here because the webhook APIs should be processed in a different manner compared to websocket APIs. And let's proceed with this for the moment.